test/integration/instances_test.go

```diff
@@ -589,6 +590,167 @@ func TestInstance_withPG(t *testing.T) {
 	require.Equal(t, inst.PlacementGroup.PlacementGroupPolicy, pg.PlacementGroupPolicy)
 }
 
+func TestInstance_CreateWithReservedIPAddress(t *testing.T) {
+	client, teardown := createTestClient(t, "fixtures/TestInstance_CreateWithReservedIPAddress")
+	defer teardown()
+
+	// Reserve an IP for testing
+	reservedIP, err := client.ReserveIPAddress(context.Background(), linodego.ReserveIPOptions{Region: "us-east"})
+	if err != nil {
+		t.Fatalf("Failed to reserve IP: %v", err)
+	}
+	defer func() {
+		err := client.DeleteReservedIPAddress(context.Background(), reservedIP.Address)
+		if err != nil {
+			t.Errorf("Failed to delete reserved IP: %v", err)
+		}
+	}()
+
+	instance, instanceTeardown, err := createInstanceWithReservedIP(t, client, reservedIP.Address)
+	if err != nil {
+		t.Fatalf("Error creating instance with reserved IP: %s", err)
+	}
+	defer instanceTeardown()
+
+	fmt.Println(instance.IPv4)
+
+}
+
+func createInstanceWithReservedIP(
+	t *testing.T,
+	client *linodego.Client,
+	reservedIP string,
+	modifiers ...instanceModifier,
+) (*linodego.Instance, func(), error) {
+	t.Helper()
+
+	createOpts := linodego.InstanceCreateOptions{
+		Label:    "go-test-ins-reserved-ip-" + randLabel(),
+		Region:   "us-east",
+		Type:     "g6-nanode-1",
+		Booted:   linodego.Pointer(false),
+		Image:    "linode/alpine3.17",
+		RootPass: randPassword(),
+		Interfaces: []linodego.InstanceConfigInterfaceCreateOptions{
+			{
+				Purpose:     linodego.InterfacePurposePublic,
+				Label:       "",
+				IPAMAddress: "",
+			},
+		},
+		Ipv4: []string{reservedIP},
+	}
+
+	for _, modifier := range modifiers {
+		modifier(client, &createOpts)
+	}
+
+	instance, err := client.CreateInstance(context.Background(), createOpts)
+	if err != nil {
+		return nil, func() {}, err
+	}
+
+	teardown := func() {
+		if terr := client.DeleteInstance(context.Background(), instance.ID); terr != nil {
+			t.Errorf("Error deleting test Instance: %s", terr)
+		} else {
+			fmt.Println("Deleted Linode!")
+		}
+	}
+
+	return instance, teardown, nil
+}
+
+func TestInstance_CreateWithReservedIPAddressVariants(t *testing.T) {
+	client, teardown := createTestClient(t, "fixtures/TestInstance_CreateWithReservedIPAddressVariants")
+	defer teardown()
+
+	// Reserve an IP for testing
+	reservedIP, err := client.ReserveIPAddress(context.Background(), linodego.ReserveIPOptions{Region: "us-east"})
+	if err != nil {
+		t.Fatalf("Failed to reserve IP: %v", err)
+	}
+	defer func() {
+		err := client.DeleteReservedIPAddress(context.Background(), reservedIP.Address)
+		if err != nil {
+			t.Errorf("Failed to delete reserved IP: %v", err)
+		}
+	}()
+
+	// Test: Owned non-assigned reserved IP
+	instance, instanceTeardown, err := createInstanceWithReservedIP(t, client, reservedIP.Address)
```

Comment:
Consider splitting this test case into a test case per what you are testing. You may then not need to print success messages, and the fixtures would become smaller files, making it easier to look into them in case of failure.

Reply:
Test case has been split into individual test functions.